projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php

```diff
@@ -760,3 +770,47 @@ function wpcom_global_styles_is_previewing_premium_theme_without_premium_plan( $
 
 	return ! $has_premium_plan_or_higher;
 }
+
+/**
+ * Checks whether the site has access to Global Styles with a Personal plan as part of an A/B test.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has access to Global Styles with a Personal plan.
+ */
+function wpcom_site_has_global_styles_in_personal_plan( $blog_id = 0 ) {
+	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+		return false;
+	}
+
+	if ( ! function_exists( '\ExPlat\assign_given_user' ) ) {
+		return false;
+	}
+
+	if ( ! $blog_id ) {
+		$blog_id = get_current_blog_id();
+	}
+
+	$cache_key                          = "global-styles-on-personal-feb-2025-$blog_id";
+	$found_in_cache                     = false;
+	$has_global_styles_in_personal_plan = wp_cache_get( $cache_key, 'a8c_experiments', false, $found_in_cache );
```

Comment:
Out of curiosity, why do we need to cache it? 🤔 Have you notice a performance problem on Simple without caching?

I'm asking this because for RDVs we don't have a cache and I was wondering if we need one.

Reply:
When I looked up the previous GS experiment we had caching. It makes sense since we're not making the call to explat for assigning the user again.